Sources/ParserCommand.swift

```diff
@@ -0,0 +1,132 @@
+)
+
+let paramsOption = VariadicOption<String>(
+  "param", [],
+  description: "List of template parameters"
+)
+
+struct ParserCommand {
+  let parser: Parser.Type
+  let command: CommandType
+
+  init(parser: Parser.Type) {
+    self.parser = parser
+    self.command = Commander.command(
+      outputOption,
+      templateNameOption,
+      templatePathOption,
+      paramsOption,
+      VariadicArgument<Path>("PATH", description: parser.commandInfo.pathDescription, validator: pathsExist)
+    ) { output, templateName, templatePath, parameters, paths in
+
+      try ParserCommand.execute(parserType: parser,
+                                output: output,
+                                templateName: templateName,
+                                templatePath: templatePath,
+                                parameters: parameters,
+                                paths: paths)
+    }
+  }
+
+  init(parser: Parser.Type,
+       deprecatedOption: Option<String>,
+       deprecatedHandler: @escaping (String) throws -> Void) {
+    self.parser = parser
+    self.command = Commander.command(
+      outputOption,
+      templateNameOption,
+      templatePathOption,
+      paramsOption,
+      deprecatedOption,
+      VariadicArgument<Path>("PATH", description: parser.commandInfo.pathDescription, validator: pathsExist)
+    ) { output, templateName, templatePath, parameters, deprecated, paths in
+
+      try deprecatedHandler(deprecated)
+
+      try ParserCommand.execute(parserType: parser,
+                                output: output,
+                                templateName: templateName,
+                                templatePath: templatePath,
+                                parameters: parameters,
+                                paths: paths)
+    }
+  }
+
+  init(parser: Parser.Type,
+       deprecatedOption1: Option<String>,
+       deprecatedOption2: Option<String>,
+       deprecatedHandler: @escaping (String, String) throws -> Void) {
+    self.parser = parser
+    self.command = Commander.command(
+      outputOption,
+      templateNameOption,
+      templatePathOption,
+      paramsOption,
+      deprecatedOption1,
+      deprecatedOption2,
+      VariadicArgument<Path>("PATH", description: parser.commandInfo.pathDescription, validator: pathsExist)
+    ) { output, templateName, templatePath, parameters, deprecated1, deprecated2, paths in
+
+      try deprecatedHandler(deprecated1, deprecated2)
+
+      try ParserCommand.execute(parserType: parser,
+                                output: output,
+                                templateName: templateName,
+                                templatePath: templatePath,
+                                parameters: parameters,
+                                paths: paths)
+    }
+  }
+
+  // swiftlint:disable:next function_parameter_count
+  private static func execute(parserType: Parser.Type,
```

Comment:
The reason I made this static, is because `ParserCommand` is a value type and can't reference itself in the init (I think, it's been a while since I wrote this). Should we just make it a reference type to make this easier? There'll be no need to pass along the `parserType`.